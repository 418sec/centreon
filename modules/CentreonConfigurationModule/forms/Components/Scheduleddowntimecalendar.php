<?php
/*
 * Copyright 2005-2015 CENTREON
 * Centreon is developped by : Julien Mathis and Romain Le Merlus under
 * GPL Licence 2.0.
 *
 * This program is free software; you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation ; either version 2 of the License.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
 * PARTICULAR PURPOSE. See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program; if not, see <http://www.gnu.org/licenses>.
 *
 * Linking this program statically or dynamically with other modules is making a
 * combined work based on this program. Thus, the terms and conditions of the GNU
 * General Public License cover the whole combination.
 *
 * As a special exception, the copyright holders of this program give CENTREON
 * permission to link this program with independent modules to produce an executable,
 * regardless of the license terms of these independent modules, and to copy and
 * distribute the resulting executable under terms of CENTREON choice, provided that
 * CENTREON also meet, for each linked independent module, the terms  and conditions
 * of the license of that module. An independent module is a module which is not
 * derived from this program. If you modify this program, you may extend this
 * exception to your version of the program, but you are not obliged to do so. If you
 * do not wish to do so, delete this exception statement from your version.
 *
 * For more information : contact@centreon.com
 *
 */

namespace CentreonConfiguration\Forms\Components;

use Centreon\Internal\Di;
use Centreon\Internal\Form\Component\Component;

/**
 * Component for select the period
 *
 * @author Maximilien Bersoult <mbersoult@centreon.com>
 * @package CentreonConfiguration
 * @subpackage Form
 * @version 3.0.0
 */
class Scheduleddowntimecalendar extends Component
{
    /**
     * Generate and return the html for the element
     *
     * @param array $element The element to parse
     * @return array
     */
    public static function renderHtmlInput(array $element)
    {
        $tpl = Di::getDefault()->get('template');
        $weeklyDays = array(
            0 => "sunday",
            1 => "monday",
            2 => "tuesday",
            3 => "wednesday",
            4 => "thursday",
            5 => "friday",
            6 => "saturday"
        );

        $tpl->assign('element', $element);
        $tpl->assign('weeklyDays', $weeklyDays);
        $tpl->addCss('centreon.scheduled-downtime.css', 'centreon-configuration');
        $tpl->addCss('bootstrap-datetimepicker.min.css');
        $tpl->addJs('centreon.scheduled-downtime.js', 'bottom', 'centreon-configuration');
        $tpl->addJs('hogan-3.0.0.min.js');
        $tpl->addJs('bootstrap-datetimepicker.min.js');
         
        return array(
            'html' => $tpl->fetch('file:[CentreonConfigurationModule]/forms/components/scheduled_downtime_calendar.tpl'),
        );
    }
}
