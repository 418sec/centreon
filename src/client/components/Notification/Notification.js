import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from 'material-ui/styles'
import Snackbar from 'material-ui/Snackbar'
import CloseIcon from 'material-ui-icons/Close'
import IconButton from 'material-ui/IconButton'

const styles = theme => ({
  information: {
    backgroundColor: '#00bfb3'
  },
  warning: {
    backgroundColor: '#ff9a13'
  },
  error: {
    backgroundColor: '#e00b3d'
  }
})

const Notification = ({ notifications, onClose, classes }) => (
  <div>
    {Object.keys(notifications).map((id, index) => {
      if (index === 0) {
        return (
          <Snackbar
            key={'notification-' + id}
            open={true}
            onClose={(event, reason) => {
              if (reason !== 'clickaway') {
                onClose(id)
              }
            }}
            autoHideDuration={notifications[id].timeout}
            anchorOrigin={{
              vertical: notifications[id].position.vertical,
              horizontal: notifications[id].position.horizontal
            }}
            SnackbarContentProps={{
              className: classes[notifications[id].type]
            }}
            message={notifications[id].message}
            action={[,
              <IconButton
                key="close"
                id="closeNotification"
                aria-label="Close"
                color="inherit"
                onClick={() => {onClose(id)}}
              >
                <CloseIcon />
              </IconButton>,
            ]}
          />
        )
      }
    })}
  </div>
)

Notification.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
};

export default withStyles(styles, { withTheme: true })(Notification)