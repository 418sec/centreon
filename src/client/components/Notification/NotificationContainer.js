import React, { Component } from 'react'
import { connect } from 'react-redux'
import Notification from './Notification'
import { closeNotification } from 'Redux/notification/notificationActions'

class NotificationContainer extends Component {
  render = () => {
    const { notifications, handleClose } = this.props

    return (
      <Notification
        notifications={notifications}
        onClose={handleClose}
      />
    )
  }
}

const mapStateToProps = (store) => {
  return {
    notifications: store.notificationReducer.notifications,
  }
}


const mapDispatchToProps = (dispatch) => {
  return {
    handleClose: (id) => {
      return dispatch(closeNotification(id))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(NotificationContainer)