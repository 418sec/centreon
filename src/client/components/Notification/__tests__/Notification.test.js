/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { Notification } from 'Components/Notification'
import configureStore from 'redux-mock-store'

Enzyme.configure({ adapter: new Adapter() })

describe('Notification', () => {
  const mockStore = configureStore()
  const store = mockStore({
    notificationReducer: {
      notifications: {
        '1': {
          type: 'information',
          message: 'test notification',
          position: {
            vertical: 'bottom',
            horizontal: 'right'
          },
          timeout: 3000
        }
      }
    }
  })
  let component

  it('renders correctly', () => {
    component = shallow(
      <Notification
        store={store}
      />
    ).dive().dive().dive()
    expect(toJson(component)).toMatchSnapshot()
  })

  it('closes correctly', () => {
    let actions

    component = mount(
      <Notification
        store={store}
      />
    )
    component.find('button#closeNotification').simulate('click')
    actions = store.getActions()
    expect(actions[0].type).toBe('CLOSE_NOTIFICATION')
  })
})