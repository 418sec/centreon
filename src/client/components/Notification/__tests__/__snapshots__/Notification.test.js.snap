// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders correctly 1`] = `
<div>
  <WithStyles(Snackbar)
    SnackbarContentProps={
      Object {
        "className": "Notification-information-1",
      }
    }
    action={
      Array [
        undefined,
        <WithStyles(IconButton)
          aria-label="Close"
          color="inherit"
          id="closeNotification"
          onClick={[Function]}
        >
          <pure(Close) />
        </WithStyles(IconButton)>,
      ]
    }
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "bottom",
      }
    }
    autoHideDuration={3000}
    key="notification-1"
    message="test notification"
    onClose={[Function]}
    open={true}
  />
</div>
`;
